Http client

yassiezar opened the pull request:
┆Issue is synchronized with this [Jira Task](https://robosoft-ai.atlassian.net/browse/SMACC2-176) by [Unito](https://www.unito.io)

yassiezar committed:
Ported http work over from smacc1

yassiezar committed:
Ported http work over from smacc1

yassiezar committed:
Added example showasing http requests

yassiezar committed:
Added example showasing http requests

yassiezar committed:
Fixed some typos in the HTTP example

yassiezar committed:
Fixed some typos in the HTTP example

yassiezar committed:
Made provision for HTTP GET/POST in the HTTP client

yassiezar committed:
Made provision for HTTP GET/POST in the HTTP client

brettpac commented:
Hey @yassiezar. Is there any issue with moving the client folder https://github.com/yassiezar/SMACC2/tree/http-client/smacc2_sm_reference_library/sm_atomic_http/include/sm_atomic_http/clients/client_behaviors to the https://github.com/robosoft-ai/SMACC2/tree/humble/smacc2_client_library folder?

yassiezar commented:
No issues,  but I'll probably need to modify it to make it a bit more generic so that its useful to other users

brettpac commented:
Are there any complications with...

    Moving the client https://github.com/yassiezar/SMACC2/blob/http-client/smacc2/include/smacc2/client_bases/smacc_http_client.hpp, to it's own folder in the smacc_client_library folder?
    Moving the code to a cpp file?

    Let's chat offline about this.

yassiezar committed:
Moved HTTP client into the client_libraries folder and made separate compilation units. Modified examples to use this new client

yassiezar committed:
Moved HTTP client into the client_libraries folder and made separate compilation units. Modified examples to use this new client

yassiezar commented:
@brettpac I moved the packages and made separate CPP files. I still want to modify the client to be more generically useful

brettpac committed:
Update smacc_action_client_base.hpp

Pretty sure this change was made accidently.

brettpac committed:
Update smacc_action_client_base.hpp

Messed up first commit

brettpac commented on smacc2_sm_reference_library/sm_atomic_http/include/sm_atomic_http/clients/client_behaviors/cb_http_request.hpp:
The only thing left to do is move this file into a client behaviors folder in the http client directory.

brettpac committed:
Update http_client.cpp

Added License to file to clean up format error

brettpac committed:
Update http_client.cpp

Added License to file to clean up format error

brettpac committed:
Update http_client.cpp

Found better Copyright & License for CI Format error.

brettpac committed:
Update http_client.cpp

Found better Copyright & License for CI Format error.

brettpac committed:
Update cb_http_request.hpp

Another little format error

brettpac committed:
Update cb_http_request.hpp

Another little format error

brettpac committed:
Update sm_atomic_http.py

Somehow a bracket became a cNode

brettpac committed:
Update sm_atomic_http.py

Somehow a bracket became a cNode

yassiezar commented:
@brettpac just to update you, the example isn't working as expected anymore and HTTP responses aren't being received. I'm busy investigating, but I suspect it has to do with changes to Boost::beast's APIs. I'll ping you when its resolved

yassiezar committed:
- Fixed errors with HTTP requests not triggering,
- Fixed response callback not triggering correctly and causing a crash
- Fixed example not transitioning back to state 1 after a response has been received

yassiezar committed:
- Fixed errors with HTTP requests not triggering,
- Fixed response callback not triggering correctly and causing a crash
- Fixed example not transitioning back to state 1 after a response has been received

yassiezar committed:
Added helper class to handle server name related interactions

yassiezar committed:
Added SSL implementation, need to make provision for SSL and non-SSLD requests

yassiezar committed:
Added SLL certs

yassiezar committed:
Split the HTTp session class into own compile unit

yassiezar committed:
Added HTTP session base class and implemented the SSL version. The client can decide between which session to initialise at runtime depending on the server set

yassiezar committed:
Implemented non-ssl HTTP session for clients

brettpac commented:
I think it's just being held up by precommit... 

Command: pre-commit run -a

brettpac commented:
Hi @yassiezar , thank you for this pull request. Looks awesome. 
One last request... 

For the cb_http_request found here...
[smacc2_sm_reference_library/sm_atomic_http/include/sm_atomic_http/clients/client_behaviors](smacc2_sm_reference_library/sm_atomic_http/include/sm_atomic_http/clients/client_behaviors)

Let's move that into a new folder entitled "client_behaviors" and place that folder inside of this one...
[smacc2_client_library/http_client/include/http_client/](smacc2_client_library/http_client/include/http_client)

So the end result would look like...
smacc2_client_library/http_client/include/http_client/client_behaviors/cb_http_request.hpp

Thank you,

yassiezar committed:
Removing hard-coded ssl certs and using the default OpenSSL certs for
verification

yassiezar committed:
Run through formatter

yassiezar committed:
Merged in with changes from ssl

yassiezar committed:
Fix bad merge

brettpac commented:
Hi Jaycee, looks like we're missing some copyrights...
 smacc2_client_library/http_client/src/http_client/http_session.cpp: could not find copyright notices...
```
1 errors, checked 306 files
smacc2_client_library/http_client/src/http_client/ssl_http_session.cpp: could not find copyright notice
1 errors, checked 296 files
smacc2_client_library/http_client/include/http_client/ssl_http_session.hpp: could not find copyright notice
1 errors, checked 274 files
smacc2_client_library/http_client/include/http_client/http_session.hpp: could not find copyright notice
smacc2_client_library/http_client/include/http_client/http_session_base.hpp: could not find copyright notice
```

brettpac commented:
// Copyright 2021 RobosoftAI Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/*****************************************************************************************************************
*

    Author: Jacobus Lock

******************************************************************************************************************/

yassiezar committed:
Merge branch 'humble' into http-client

yassiezar committed:
Remove noisy changelog

yassiezar committed:
Undo change that introduced and error

brettpac commented:
Hmmm. As Im reading it, the failed binary check is tied to the sm_simple_action_client state machine package I just merged a few minutes ago...

yassiezar committed:
Added missing copyrights

yassiezar commented:
@brettpac I just merged in your changes. That might fix the error

yassiezar commented on .pre-commit-config.yaml:
```diff
@@ -63,7 +63,7 @@ repos:
         name: clang-format
         description: Format files with ClangFormat.
-        entry: clang-format-12
+        entry: clang-format
```
@brettpac I replaced the usage of the specific clang-format version with the generic executable. Currently Ubuntu installs clang-format 14

yassiezar replied:
nevermind, I see the CI container is specifically looking for clang-format-12

yassiezar committed:
Revert change to precommit config

yassiezar committed:
Fixing copyright, formatting

yassiezar committed:
Fixed whitespace blocking precommit

yassiezar committed:
Revert change to precommit config

brettpac commented:
I think to get this passing we just have to run precommit twice. Once to make fixes (which I can see it does from the logs), and the other to approve.

yassiezar commented:
All the checks pass when i run it locally, not sure how to get it to pass here. I'm not sure why the `clang-format` check is failing above, does the command need to be edited to use `clang-format` instead of `clang-format-12`? Ubuntu 22.04 uses `clang-format-14`

yassiezar committed:
Merge branch 'humble' into http-client
